src/modules/cmdpal/Microsoft.CmdPal.UI.ViewModels/TopLevelCommandManager.cs

```diff
@@ -81,19 +81,28 @@ private async Task<IEnumerable<TopLevelViewModel>> LoadTopLevelCommandsFromProvi
 
         await commandProvider.LoadTopLevelCommands(_serviceProvider, weakSelf);
 
-        var settings = _serviceProvider.GetService<SettingsModel>()!;
-
-        List<TopLevelViewModel> commands = [];
+        var commands = await Task.Factory.StartNew(
+            () =>
+            {
+                List<TopLevelViewModel> commands = [];
+                foreach (var item in commandProvider.TopLevelItems)
+                {
+                    TopLevelCommands.Add(item);
+                }
 
-        foreach (var item in commandProvider.TopLevelItems)
-        {
-            commands.Add(item);
-        }
+                foreach (var item in commandProvider.FallbackItems)
+                {
+                    if (item.IsEnabled)
+                    {
+                        TopLevelCommands.Add(item);
```

Comment:
This doesn't look right: modifies collection directly and circumvents the lock on TopLevelCommands

```suggestion
                        commands.Add(item);
```